src/structures/ComponentInteraction.js

```diff
@@ -0,0 +1,233 @@
+'use strict';
+
+const APIMessage = require('./APIMessage');
+const Interaction = require('./Interaction');
+const WebhookClient = require('../client/WebhookClient');
+const { Error } = require('../errors');
+const { InteractionResponseTypes } = require('../util/Constants');
+const MessageFlags = require('../util/MessageFlags');
+
+/**
+ * Represents a message button interaction.
+ * @extends {Interaction}
+ */
+class ComponentInteraction extends Interaction {
```

Comment:
This class should be exported.

Reply:
It is. Line 233

Reply:
Yes, from index, so you can import this from outside of discord.js.